core/src/main/java/com/linecorp/armeria/client/retry/AbstractRetryingClient.java

```diff
@@ -70,42 +70,68 @@
             AttributeKey.valueOf(AbstractRetryingClient.class, "STATE");
 
     @Nullable
-    private final RetryStrategy retryStrategy;
+    private final RetryRule retryRule;
 
     @Nullable
-    private final RetryStrategyWithContent<O> retryStrategyWithContent;
+    private final RetryRuleWithContent<O> retryRuleWithContent;
 
     private final int maxTotalAttempts;
     private final long responseTimeoutMillisForEachAttempt;
 
     /**
      * Creates a new instance that decorates the specified {@link Client}.
      */
+    protected AbstractRetryingClient(Client<I, O> delegate, RetryRule retryRule,
```

Comment:
Could be package-local?

Reply:
OK, then will guide the deprecated constructor to use the builder of RetryingClient or RetryRpcClient?

Reply:
Yes!